dependency/dependency.go

```diff
@@ -49,6 +49,14 @@ const (
 	DefaultContextTimeout = 60 * time.Second
 )
 
+func (t Type) String() string {
+	if t > 2 {
```

Comment:
minor: could we have a comment here indicating why `2` or use a named constant?

Reply:
It was used in the upstream PR to default to `unknown` probably because `Nomad` was not part of the `Type` list at that time.
I replaced this section by a switch case, more explicit